lib/hooks/useResources.ts

```
@@ -0,0 +1,200 @@
+  // Listen for resource events using DOM events
+  useEffect(() => {
+    if (!isSignedIn) return;
+
+    // Event handler for resource changes (created, updated, deleted)
+    const handleResourceChanged = async () => {
+      try {
+        // Fetch all modules for the selector without showing loading state
+        const modulesData = await api.getModules();
+        const modulesList = modulesData.modules || [];
+        setResourceModules(
+          modulesList.map((m: any) => ({
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)